### PR TITLE
Add service for acknowledging subscriptions (#48)

### PR DESCRIPTION
**Pull request by iDuuck**

### CODE REVIEW

**Comment by ToTenMilan** [comment]
Thank you @iDuuck and candy_check team for your great work. Do you plan to merge this PR anytime soon?

**Comment by iDuuck** [comment]
@ToTenMilan I am currently using this PR myself in a production application/API. You can use the PR instead of the current `candy_check` gem using the following line(s) in your Gemfile:

```ruby
gem 'candy_check', git: 'https://github.com/use-pupy/candy_check.git',
                   branch: 'feature/subscription_acknowledgement'
```

---

**Disclaimer:** While I am in no will of spoofing anything, it is a general disclaimer, that you should *always* be sure, to check the software you are installing on your system/software. So be sure to check the branch out and what was edited and only then install it into your software.

**Comment by courtsimas** [comment]
@jnbt any chance this is worthy of being pulled in? I would love it in the primary gem instead of using a forked version.

**Comment by iDuuck** [comment]
I hope it will be merged at some point. For now using my branched version in a production-ready application.

**Comment by courtsimas** [comment]
@iDuuck Since the official candy_check gem already does acknowledgments and you can pass in a subscription id, like
```
result = acknowledger.acknowledge_product_purchase(
  package_name: "my-package-name",
  product_id: "my-subscription-id",
  token: "my-token"
)
```

(according to the readme),... then what does this PR provide that's different?  I'm naive and currently implementing IAP for iOS, and about to dive into google/android land so I'm just curious.

**Comment by iDuuck** [comment]
@courtsimas So, actually while it **might** work out (Spoiler: It didn't for me.) for some time, there could be a place in time, where the subscription will not be acknowledged anymore, because Google might cancel the ability to use the product acknowledgement endpoint for the subscriptions.

Looking at the Google Play API's and the Ruby Library this library is using, you are seeing, that there are actually two different methods for acknowledgment ([Product ack.](https://googleapis.dev/ruby/google-api-client/latest/Google/Apis/AndroidpublisherV3/AndroidPublisherService.html#acknowledge_purchase_product-instance_method) & [Subscription ack.](https://googleapis.dev/ruby/google-api-client/latest/Google/Apis/AndroidpublisherV3/AndroidPublisherService.html#acknowledge_purchase_subscription-instance_method)).

TLDR: This being said, I think, the endpoint (and the methods) might work for you, however, I would not rely my production code on those.

**Comment by courtsimas** [comment]
Fair enough. Makes sense. Like I mentioned, I haven’t done this yet - just now diggin in and planning my route. This is good to know.

**Comment by fabn** [comment]
@jnbt this would be a great addition to this gem, could you please take a look at it?

Thanks.

**Comment by iDuuck** [comment]
Very sad to still see this open...

**Review by christophweegen — CHANGES_REQUESTED** [review]
Hello @iDuuck thanks for your contribution and sorry for the very late feedback.

Could you maybe please add a short section in the `README` with an example so this functionality is documented somewhere?

Once this is done this should be ready to be merged.

@jnbt what do you think?

**Comment by iDuuck** [comment]
Hey @christophweegen, would be awesome to get the opinion of @jnbt first, before putting more effort into the PR.

**Comment by jnbt** [comment]
@iDuuck I agree. Everything looks fine. Just some instructions how to use it, would be really useful.

**Comment by alhassanreem** [comment]
@iDuuck @jnbt Any update on this merge, it would be really useful to have this on the main gem

**Comment by mike927** [comment]
@jnbt Are there any obstacles to merge this PR? I would love to have to as acknowledging subscriptions is crucial part and it's missing now.

**Comment by iduuck** [comment]
I am not using the library anymore actively, so I am not very eager to write a doc on this one. Would be awesome to get the docs written by someone after digging into the code here.
